### ISSUE TITLE
Push controller allowed to start multiple times

### ISSUE THREAD

**Issue by wtong98** [open-telemetry/opentelemetry-go#998]
The Push controller is structured such that `Start()` may be called multiple times (see [here](https://github.com/open-telemetry/opentelemetry-go/blob/fd61d2edece005195116e6b9218801a460a3868d/sdk/metric/controller/push/push.go#L95)), each launching a goroutine that initiates its own collection.

The behavior seems to be somewhat expected, as there exists a wait group to block while multiple collection goroutines are executing, and [a test](https://github.com/open-telemetry/opentelemetry-go/blob/fd61d2edece005195116e6b9218801a460a3868d/sdk/metric/controller/push/push_test.go#L134) that checks that starting multiple collection routines is ok.

However, having multiple collection routines seems somewhat redundant. What if instead we disallow multiple starts, and enforce only a single running routine?

**Comment by jmacd** [comment]
It looks like the push controller's Start() method will only start one goroutine, and the test is just checking that it doesn't crash, for example, and that Stop() still works if you happen to start more than once.

It's this code:
```
// Start begins a ticker that periodically collects and exports
// metrics with the configured interval.
func (c *Controller) Start() {
	c.lock.Lock()
	defer c.lock.Unlock()

	if c.ticker != nil {
		return
	}

	c.ticker = c.clock.Ticker(c.period)
	c.wg.Add(1)
	go c.run(c.ch)
}
```

Perhaps we could just add a comment saying that multiple starts are ignored?

**Comment by wtong98** [comment]
My apologies, overlooked the if condition!

Would the wait group still be needed, if only one goroutine is allowed to run at a time?

**Comment by jmacd** [comment]
I believe the wait group ensures that `run()` exits before `tick()` is called one last time. (Noyr the body of `run` is not protected by a lock.)

**Comment by jmacd** [comment]
Will fix this in #1378 (was suggested by reviewer).
